### ISSUE TITLE
Can't get env when submodules share names with types

### ISSUE THREAD

**Issue by scolsen** [carp-lang/Carp#875]
```
carp: Can't get env.
CallStack (from HasCallStack):
  error, called at src/Lookup.hs:217:34 in CarpHask-0.3.0.0-JeTRwcTheISKNh9w0jc5KF:Lookup

```

**EDIT**: This seems to happen when a type shares a name with a submodule:

```
(deftype Foo Bar)

(defmodule Baz 
  (defmodule Foo
   ........
))
```

**Comment by hellerve** [comment]
This is interesting. I can’t reproduce this bug with the following code:

```clojure
(deftype Foo Bar)

(defmodule Baz
  (defmodule Foo
    (defn x [] 1)))
```

But there is in fact a bug there, maybe different. If this is just put in the REPL, it seems to work at first. But then, if we try to `(build)`, we get:

```
carp: Can't insert into non-existing module: Foo
CallStack (from HasCallStack):
  error, called at src/Lookup.hs:192:16 in CarpHask-0.3.0.0-JeTRwcTheISKNh9w0jc5KF:Lookup
```

Which I presume is the same bug rearing a different, equally ugly head.

**Comment by hellerve** [comment]
Actually, this looks like it’s related to #865 as well.

**Comment by scolsen** [comment]
> This is interesting. I can’t reproduce this bug with the following code:
> 
> 
> 
> ```clojure
> 
> (deftype Foo Bar)
> 
> 
> 
> (defmodule Baz
> 
>   (defmodule Foo
> 
>     (defn x [] 1)))
> 
> ```
> 
> 
> 
> But there is in fact a bug there, maybe different. If this is just put in the REPL, it seems to work at first. But then, if we try to `(build)`, we get:
> 
> 
> 
> ```
> 
> carp: Can't insert into non-existing module: Foo
> 
> CallStack (from HasCallStack):
> 
>   error, called at src/Lookup.hs:192:16 in CarpHask-0.3.0.0-JeTRwcTheISKNh9w0jc5KF:Lookup
> 
> ```
> 
> 
> 
> Which I presume is the same bug rearing a different, equally ugly head.

oh wow! yeah I should have mentioned the bug in the op occurs when you `load` a source with the submodule setup

**Comment by hellerve** [comment]
Oh, this is getting quite complicated. I think it boils down to modules that already exist in a parent environment being problematic.

**Comment by eriksvedang** [comment]
I fail to reproduce this...

**Comment by scolsen** [comment]
If I use this file:

```clojure
;; foo.carp

(deftype Foo Bar)

(defmodule Baz
  (defmodule Foo
    (defn foo [] 0)
  )
)
```

and call `carp -x foo.carp`

I get:

```
carp: Can't insert into non-existing module: Foo
CallStack (from HasCallStack):
  error, called at src/Env.hs:22:16 in CarpHask-0.4.2.0-K2sdzDEAjn25KRaWCaWwhV:Env
```

If I `load` it in the repl it works until I attempt to use `Baz` in any way:

```clojure
鲤 (load "foo.carp")
鲤 :i Baz
carp: Can't insert into non-existing module: Foo
CallStack (from HasCallStack):
  error, called at src/Env.hs:22:16 in CarpHask-0.4.2.0-K2sdzDEAjn25KRaWCaWwhV:Env
```

**Comment by eriksvedang** [comment]
@scolsen OK, thanks!

**Comment by scolsen** [comment]
fixed in #1078
